Package.swift

```diff
@@ -6,21 +6,23 @@ import PackageDescription
 let package = Package(
     name: "FTAPIKit",
     products: [
-        // Products define the executables and libraries produced by a package, and make them visible to other packages.
         .library(
             name: "FTAPIKit",
-            targets: ["FTAPIKit"])
+            targets: ["FTAPIKit"]),
+        .library(
```

Comment:
Od SwiftPM 5 můžeš přidávat "define" s podmínkama pro Swift package (viz https://github.com/apple/swift-package-manager/blob/master/Documentation/PackageDescription.md#methods-8). Raději bych využil tuhle možnost než "FTAPIKitPromises".

Výhody těch flagů jsou podle mě zásadní. Celé to bude "horizontální" tj. když přidáme další "subspecy" tak nebudeme muset vytvářet target s několikaslovným názvem pro kažodu variantu. Dál přidáním maker umožňuješ resolveru importovat tu knihovnu jednou, takhle by se pro každý subspec imprtovala znova (a to ani nemluvím o tom, že by to mohlo rozbít namespace). 

_Tohle není blokující review, jenom ta věc nahoře._

Reply:
Určitě to teď předělávat nebudu.

- Nemyslím si, že je to korektní způsob jak to dělat.
- Importování stejné knihovny víckrát nevadí, protože ten dynamický framwork přidá do aplikace právě jednou.
- Nevím jak tímto způsobem popsat závisloti.
- Určitě to můžeme prozkoumat, ale tohle řešení spolehlivě funguje a není z mojí hlavy.

Reply:
- Importování stejné knihovny víckrát nevadí, protože ten dynamický framwork přidá do aplikace právě jednou.

Máš pravdu.

Každopádně určitě bych souhlasil s tím posledním bodem. Pokud jsem ale správně pochopil https://github.com/apple/swift-evolution/blob/master/proposals/0238-package-manager-build-settings.md#motivation tak tohle je přesně důvod přidání té funkce. 

_Proto říkám že to neberu jako breaking, sám jsem submoduly viděl, někteří se uchylovali ke kombinacím maker a importování vzdálených balíků. Jiní prostě jen omakrovali seznam modulů. Byl to mess. Pokud bych měl zvolit způsob jak dělat submoduly, udělal bych to rozhodně pomocí toto a explicitních include/exclude. https://github.com/apple/swift-package-manager/blob/master/Documentation/PackageDescription.md#methods-5 Ale připadá mi obecně že tohle není dořešené. Můžu se na to podívat jestli budeš chtít._